Support large github organisations

skwashd opened the pull request:
The current github integration only retrieves the first page of results when looking up user's team membership. For most organisations this isn't a problem. When you have a large organisation with over 100 teams not all the teams are retrieved. This patch solves this problem by doing the following things:

* Modify HttpGet() return to include the response headers
* Look up _all_ the teams the user is a member of

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/grafana/grafana?pullRequest=8846) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/grafana/grafana?pullRequest=8846) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/grafana/grafana?pullRequest=8846) it.</sub>

marji commented:
I have tested this change and it works.

1) I had grafana 4.3.2 installed, and had trouble with github authentication - some users from the same team and organisation could not log in.
2) I built grafana from this PR and copied over the new grafana-server binary over the existing installation 
3) restarted grafana-server and users can now login with no problems.

torkelo commented on pkg/social/common.go:
```diff
@@ -23,24 +23,26 @@ func isEmailAllowed(email string, allowedDomains []string) bool {
 }
 
-func HttpGet(client *http.Client, url string) ([]byte, error) {
+func HttpGet(client *http.Client, url string) ([]byte, http.Header, error) {
```
think returning an new struct (HttpGetResponse) with the body & headers is clearer

miguelbernadi replied:
Is there a reason not to use named return values other than backwards compatibility to pre-1.7? Would remove the need to specify nil and err in all return statements.

skwashd replied:
@torkelo I'll try to get that change done over the weekend.

skwashd committed:
Support large github organisations

* Add new HttpGetResponse struct type
* Modify HttpGet() return to use HttpGetResponse
* Look up _all_ the teams the user is a member of

skwashd commented:
@torkelo I have implemented the change you requested. Given #6347 says 1.7 is the minimum version, I implemented @miguelbernadi's suggestion of using named return values.

Is there anything else you want to see done before this is merged? I have signed the CLA and the CI bot is happy.

skwashd commented:
@torkelo thanks for merging the PR. Would you consider a new patch that contains a hacked up version of the new `HttpGet()` logic in `SocialGithub.FetchTeamMemberships()` as a backport for 4.x?

torkelo commented:
Not Not i want this in stable release  branch, but master / nightly is basically the same as 4.4.x as most of our current unstable / unfinished stuff is in develop branch

skwashd commented:
Thanks for the clarification. It makes sense. For now we'll carry the patch internally against 4.4.x until 5.0 is released.
